cmd/kubelet/app/server_bootstrap_test.go

```diff
@@ -324,7 +324,7 @@ func (s *csrSimulator) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 		}
 		s.csr = csr
 
-	case req.Method == "GET" && req.URL.Path == "/apis/certificates.k8s.io/v1beta1/certificatesigningrequests" && req.URL.RawQuery == "fieldSelector=metadata.name%3Dtest-csr&limit=500&resourceVersion=0":
+	case req.Method == "GET" && req.URL.Path == "/apis/certificates.k8s.io/v1beta1/certificatesigningrequests" && req.URL.RawQuery == "fieldSelector=metadata.name%3Dtest-csr&limit=500":
```

Comment:
hmm, does this mean the first request isn't using resourceVersion=0?

Reply:
Yeah, we're deliberately switching to resourceVersion="" for initial list. I rewrote the description to explain this a bit better, but the basic idea is that we won't want reflectors to go back in time across component restarts, and since we were already already doing a quorum read for resourceVersion="0" (since we skip the watch cache with pagination), let's go ahead and get the strong guarantee that we're getting latest by using resourceVersion="" (which matters for a corner case when the pager falls back to full list).

Reply:
Huh. I **really** didn't think we were paginating initial reflector list calls. If we were, then I guess this won't regress performance, but if we weren't, how do we anticipate this changing performance for node watches of pods (high volume of watchers, across all namespaces, filtered to a small set of total results with spec.nodeName fieldSelector, is probably the worst-case scenario for the API server to pass through to etcd)

Reply:
https://github.com/kubernetes/kubernetes/blob/2bd9643cee5b3b3a5ecbd3af49d09018f0773c77/staging/src/k8s.io/apiserver/pkg/storage/cacher/cacher.go#L595-L598

The cacher ignores limit requests (which it is permitted to do) when resourceVersion=0.

Reply:
Good find @liggitt. That makes complete sense. I'll revert the initial list back to resourceVersion="0" and update the title and description of this PR to make it clear that this only prevents the reflector from going back in time for relists, not component restarts.